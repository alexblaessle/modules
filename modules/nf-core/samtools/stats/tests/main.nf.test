nextflow_process {

    name "Test Process SAMTOOLS_STATS"
    script "../main.nf"
    process "SAMTOOLS_STATS"
    tag modules
    tag modules/nf-core
    tag samtools
    tag samtools/stats

    test("Should run without failures") {

        when {
            params {
                
                outdir = "$outputDir"
            }
            process {
                """
                // define inputs of the process here.
                input[0] = [ 
                                [ id:'test', single_end:false ], // meta map
                                file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                                file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
                            ]
                """
            }
        }

        then {
            assertAll(
            {assert process.success},
            {assert snapshot(process.out).match()}

            )
        }

    }

}
