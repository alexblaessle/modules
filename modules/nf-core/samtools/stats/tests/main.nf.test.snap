{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,6e768486d5df0257351c5419a79f9c9b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,08035f3409d934d47a416150884bb0df"
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,6e768486d5df0257351c5419a79f9c9b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08035f3409d934d47a416150884bb0df"
                ]
            }
        ],
        "timestamp": "2023-10-18T11:59:06.906923"
    }
}